code/game/objects/machinery/doors/door.dm

```diff
@@ -65,8 +65,10 @@
 
 	if(ismob(AM))
 		var/mob/M = AM
-		if(world.time - M.last_bumped <= openspeed) return	//Can bump-open one airlock per second. This is to prevent shock spam.
-		M.last_bumped = world.time
+		if(M.cooldowns[COOLDOWN_BUMP]) 
+			return	//This is to prevent shock spam.
+		M.cooldowns[COOLDOWN_BUMP] = TRUE
+		addtimer(VARSET_CALLBACK(M, cooldowns[COOLDOWN_BUMP], FALSE), openspeed)
```

Comment:
Instead of `addtimer`, for pure cooldowns do something like:
`cooldowns[COOLDOWN_INDEX] = world.time + cooldown_value`
If you want to keep it short, do a macro:
`
#define COOLDOWN(index) ( cooldowns[index] > world.time )
`
Or whatever fits better your purposes.

Reply:
I specifically removed all the world.time on purpose.

Reply:
Yes, and what I'm saying is that checking `world.time` is better in this case.
I assume you removed it to keep the code smaller/cleaner. I'm suggesting a macro to achieve that.